### DISCUSSION TITLE
Using GraphSAGE for transductive learning

### DISCUSSION THREAD

**Discussion by Shiiivaa** [pyg-team/pytorch_geometric#5148] [General]
Hello,
I have a basic question about graph machine learning.
Is there any connection between the type of learning (transductive/inductive) and the type of message passing (aggregator functions like GCN or GraphSAGE) we use for graph learning?
I've seen in some papers that if you want to use inductive learning, GraphSAGE will work. While graph convolutional networks (GCNs) have only been applied in the transductive setting!

Thank you

**Comment by Padarn** [comment]
> Is there any connection between the type of learning (transductive/inductive) and the type of message passing (aggregator functions like GCN or GraphSAGE) we use for graph learning?

Warning, non-expert opinion incoming ...

The distinction is more about what you learn (generally, a feed-forward model to compute embeddings or directly learn the embeddings). I don't think this is specifically related to a message passing scheme.. but some transductive models don't use message passing, just graph walks (like DeepWalk).

Hope that's helpful.

**Comment by rusty1s** [comment]
I personally find the linkage of GraphSAGE<>inductive and GCN<>transductive super irritating, and see this differentiation more as a relict of old times. Nowadays, we group all of these different GNNs into the general concept of message passing. In particular, the concept of message passing is not really related to transductive/inductive learning. You can apply both GCN and GraphSAGE for both learning schemes.

**Reply by Padarn** [reply]
> I personally find the linkage of GraphSAGE<>inductive and GCN<>transductive super irritating

not very useful either :-)
